lib/improver/psychrometric_calculations/psychrometric_calculations.py

```diff
@@ -0,0 +1,411 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Module to contain Psychrometric Calculations."""
+
+from cf_units import Unit
+import improver.constants as constant
+import iris.analysis.maths as maths
+import numpy as np
+
+
+def check_range(cube, low, high):
+    """Function to wrap functionality for throwing out temperatures
+    too low or high for a method to use safely.
+
+    Args:
+        cube : iris.cube.Cube
+            A cube of temperature.
+
+        low : int or float
+            Lowest allowable temperature for check
+
+        high : int or float
+            Highest allowable temperature for check
+
+    Raises:
+        ValueError : If any of the values in cube.data are outside the bounds
+            set by the low and high variables.
+    """
+
+    if cube.data.max() > high or cube.data.min() < low:
+        emsg = ("This saturation vapour pressure algorithm is"
+                " only valid for temperatures between"
+                " {}K and {}K. Input cube has\n"
+                "Lowest temperature = {}\n"
+                "Highest temperature = {}")
+        raise ValueError(emsg.format(low,
+                                     high,
+                                     cube.data.min(),
+                                     cube.data.max()))
+
+
+def saturation_vapour_pressure_ashrae(temperature):
+    ''' Compute Saturation vapour pressure from temperature, using ASHRAE
+    (American Society of Heating, Refrigerating and Air-Conditioning
+    Engineers) method.
+
+    Args:
+        temp : iris.cube.Cube
+            Cube of temperature which will be converted to Kelvin
+            prior to calculation
+            Valid from -100C to 200 C
+
+    Returns:
+        result : iris.cube.Cube
+            Cube containing the saturation vapour pressure of the
+            air in Pa.
+
+    References:
+        ASHRAE Fundamentals handbook (2005) p 6.2, equation 5 and 6
+    '''
+    constant_1 = -5674.5359
+    constant_2 = 6.3925247
+    constant_3 = -0.009677843
+    constant_4 = 0.00000062215701
+    constant_5 = 2.0747825E-09
+    constant_6 = -9.484024E-13
+    constant_7 = 4.1635019
+    constant_8 = -5800.2206
+    constant_9 = 1.3914993
+    constant_10 = -0.048640239
+    constant_11 = 0.000041764768
+    constant_12 = -0.00000001445209310000
+    constant_13 = 6.5459673
```

Comment:
This could be tidied up with a dictionary or list, e.g.:
`constants = {1: -5674.5359, 2: 6.3925247, 3: -0.009677843, ...}`
`cell[...] = (np.exp(constant[1] / cell` etc

This is my preference, rather than a requirement.